### PR TITLE
PIM-6924: Navigate to complete the missing attributes

### PR DESCRIPTION
**Pull request by grena**
**Description (for Contributor and Core Developer)**

This PR contains a lot of small improvements to navigate among entities on the PEF.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Y
| Added Behats                      | Y
| Added integration tests           | N
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo

### CODE REVIEW

File: features/Context/FeatureContext.php [hunk 1]
```diff
@@ -319,6 +319,50 @@ public function fillFieldOnCurrentPage($field, $value)
...
+     *
+     *
+     * @param string $message
+     * @param string $label
+     *
+     * @throws ExpectationException
+     *
+     * @When /^I click on "(?P<message>(?:[^"]|\\")*)" footer message of the field "(?P<label>(?:[^"]|\\")*)"$/
+     */
+    public function clickOnMessageOfField($message, $label)
```
Comment (jjanvier) [inline features/Context/FeatureContext.php]: clickOnFooterMessageOfField

Comment (jjanvier) [inline features/Context/FeatureContext.php]: clickOnFooterMessageOfField ?

File: features/product-model/navigate_among_variant_entities.feature [hunk 2]
```diff
@@ -81,3 +81,13 @@ Feature: Navigate among variant entities
     Then I should see the "Rouge" element in the variant children selector for level 1
     And I should see the "Bleu" element in the variant children selector for level 1
     And completeness for element "XXL" in the variant children selector for level 1 should be "85%"
+
+  Scenario: If I can't edit an attribute that comes from a parent, I can click on a link to open the related entity
+    When I am on the "1111111127" product page
+    And I visit the "Marketing" group
+    Then I should see the text "This attribute can be updated on the attributes by Color"
```
Comment (jjanvier) [inline features/product-model/navigate_among_variant_entities.feature]: > This attribute can be updated on the attributes by Color
I don't understand the sentence oO

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/form/common/attributes.js [hunk 3]
```diff
@@ -236,6 +236,10 @@ define(
                         });
                     });
 
+                if (sessionStorage.getItem('filter_missing_required_attributes')) {
```
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/form/common/attributes.js]: different than null or false?
dunno how you do in JS

File: features/Context/FeatureContext.php [hunk 4]
```diff
@@ -319,6 +319,50 @@ public function fillFieldOnCurrentPage($field, $value)
...
+        $dropdown = $this->getCurrentPage()->getCompletenessDropdownButton();
+        $dropdown->click();
+    }
+
+    /**
+     * @When /^I click on the missing required attributes overview link$/
+     */
+    public function iClickOnTheMissingRequiredAttributesOverviewLink()
+    {
+        $link = $this->getCurrentPage()->getMissingRequiredAttributesOverviewLink();
```
Comment (juliensnz) [inline features/Context/FeatureContext.php]: does this method spin?

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/product-completeness.js [hunk 5]
```diff
@@ -160,6 +183,15 @@ define(
                     }
                 );
                 this.render();
+            },
+
+            redirectToModel: function(modelId) {
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/product-completeness.js]: jsdoc :coeuraveclesdoigts:

**Pull request merged**